includes/notifications/class.llms.notifications.php

```diff
@@ -423,19 +482,96 @@ private function name_to_file( $name ) {
 	/**
 	 * Schedule a processor to dispatch its queue on shutdown
 	 *
-	 * @param    string $type  processor name/type (eg: email)
-	 * @return   void
-	 * @since    3.8.0
-	 * @version  3.8.0
+	 * @since 3.8.0
+	 * @since [version] Use strict comparisons.
+	 *
+	 * @param string $id Processor ID (eg: email).
+	 * @return void
 	 */
-	public function schedule_processing( $type ) {
+	public function schedule_processing( $id ) {
 
-		if ( ! in_array( $type, $this->processors_to_dispatch ) ) {
+		if ( ! in_array( $id, $this->processors_to_dispatch, true ) ) {
 
-			$this->processors_to_dispatch[] = $type;
+			$this->processors_to_dispatch[] = $id;
 
 		}
 
 	}
 
+	/**
+	 * Check for processors that have items in the queue
+	 *
+	 * For any found processors, saves their queue and schedules them to be processes via a scheduled event.
+	 *
+	 * @since [version]
+	 *
+	 * @return array Array containing information about the scheduled processors.
+	 *               The array keys will be the processor ID and the values will be the timestamp of the event or a WP_Error object.
+	 */
+	public function schedule_processors_dispatch() {
+
+		$scheduled = array();
+
+		if ( $this->processors_to_dispatch ) {
+
+			foreach ( $this->processors_to_dispatch as $key => $id ) {
+
+				// Retrieve the processor.
+				$processor = $this->get_processor( $id );
+
+				// Remove it from the list of processors to dispatch.
+				unset( $this->processors_to_dispatch[ $key ] );
+
+				$scheduled[ $id ] = $processor ? $this->schedule_single_processor( $processor, $id ) : new WP_Error(
+					'invalid-processor',
+					// Translators: %s = Processor ID.
+					sprintf( __( 'The processor "%s" does not exist.', 'lifterlms' ), $id )
+				);
+
+			}
+		}
+
+		return $scheduled;
+
+	}
+
+	/**
+	 * Save pending batches and schedule the async dispatching of a processor.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Abstract_Notification_Processor $processor Notification processor object.
+	 * @param string                               $id        Processor ID.
+	 * @return int|WP_Error Timestamp of the scheduled event or an error object.
+	 */
+	protected function schedule_single_processor( $processor, $id ) {
+
+		$hook = 'llms_dispatch_notification_processor_async';
+		$args = array( $id );
+
+		// Save items in the queue.
+		$processor->save();
+
+		// Check if there's already a scheduled event.
+		$timestamp = as_next_scheduled_action( $hook, $args );
+
+		// If there's no event scheduled already, schedule one.
+		if ( ! $timestamp ) {
+
+			$timestamp = llms_current_time( 'timestamp' );
```

Comment:
Ok we have a problem here:
1) `llms_current_time( 'timestamp' )` calls `current_time( 'timestamp' )` which is _deprecated_ (it adjusts the UTC unix timestamp computed using`time()` adding the UTC offset as per WP settings. )
2) the action scheduler expects the time (`$timestamp` here) parameter to be UTC, while we're passing here the local time. This means that if in our WP settings we set the timezone to UTC+2 the action will be scheduled to be fired in 2 hours from now (I can reproduce this).

So let's go with:
`llms_current_time( 'timestamp', 1 )` that will allow us to continue testing it (which is why we have `llms_current_time` here, right?)
1) will call `current_time( 'timestamp', 1 )` that returns just `time()` and it's not _deprecated_
2) also this fixes 2
3) we all are happy

Reply:
Great catch!

Reply:
@eri-trabiccolo updated, good to merge?